go/libraries/doltcore/env/repo_state.go

```diff
@@ -44,8 +45,13 @@ type RepoStateWriter interface {
 type DocsReadWriter interface {
```

Comment:
This interface is kind of a mess. Lots of methods that do very similar things. Some of them read from disk, some of them read from previous cached information, etc.

This interface should be about one thing: moving docs file details between memory and disk. And it does way way more than that.

My first pass for what this should look like:

GetDocsOnDisk([]string docs) Docs // reads and returns docs on disk, optionally filtered by name
WriteDocsToDisk(Docs) // writes the docs given to disk

Neither of these methods involve a file system, because any implementor needs a filesystem to do its job in general. And none of them involve root values.

If you want doc details that are in a RootValue, ask the root value.
If you want to move doc details that are on disk into a RootValue, get them from the above method then tell the RootValue to update its docs table (possibly creating it in the process).
If you want update the value of the staged or working roots, there are methods in RSW for doing that.

You didn't create this mess, but it's getting really out of hand at this point. You should clean it up.

You can come back and do this on another pass.